tests/virtualization/xen/guest_management.pm

```diff
@@ -17,25 +17,36 @@ use testapi;
 use utils;
 
 sub run {
-    my ($self)     = @_;
-    my $hypervisor = get_required_var('QAM_XEN_HYPERVISOR');
-    my $domain     = get_required_var('QAM_XEN_DOMAIN');
+    select_console 'root-console';
+    opensusebasetest::select_serial_terminal();
+    my $hypervisor = get_required_var('HYPERVISOR');
+
+    record_info "REBOOT", "Reboot all guests";
+    foreach my $guest (keys %xen::guests) {
+        assert_script_run "ssh root\@$hypervisor 'virsh reboot $guest'";
+        if (script_retry("ssh root\@$hypervisor 'nmap $guest -PN -p ssh | grep open'", delay => 30, retry => 3, die => 0)) {
+            record_soft_failure "Reboot on $guest failed";
+            assert_script_run "ssh root\@$hypervisor 'virsh destroy $guest'";
+            assert_script_run "ssh root\@$hypervisor 'virsh start $guest'";
+        }
+    }
 
     record_info "SHUTDOWN", "Shut all guests down";
-    assert_script_run "ssh root\@$hypervisor 'virsh shutdown $_'" foreach (keys %xen::guests);
-    script_retry "ssh root\@$hypervisor 'virsh list --all | grep $_ | grep \"shut off\"'", delay => 3, retry => 20 foreach (keys %xen::guests);
+    foreach my $guest (keys %xen::guests) {
+        assert_script_run "ssh root\@$hypervisor 'virsh shutdown $guest'";
+        if (script_retry("ssh root\@$hypervisor 'virsh list --all | grep $guest | grep \"shut off\"'", delay => 15, retry => 6, die => 0)) {
+            record_soft_failure "Shutdown on $guest failed";
+            assert_script_run "ssh root\@$hypervisor 'virsh destroy $guest'";
+        }
+    }
 
     record_info "START", "Start all guests";
     assert_script_run "ssh root\@$hypervisor 'virsh start $_'" foreach (keys %xen::guests);
-    script_retry "ssh root\@$hypervisor 'nmap $_.$domain -PN -p ssh | grep open'", delay => 3, retry => 60 foreach (keys %xen::guests);
-
-    record_info "REBOOT", "Reboot all guests";
-    assert_script_run "ssh root\@$hypervisor 'virsh reboot $_'" foreach (keys %xen::guests);
-    assert_script_run "ssh root\@$hypervisor 'virsh list --all'";
+    script_retry "ssh root\@$hypervisor 'nmap $_ -PN -p ssh | grep open'", delay => 3, retry => 60 foreach (keys %xen::guests);
```

Comment:
One note here - you try 60 times, every 3 seconds - `60x3s`. Consider more relaxed approach, for example `12x15s`, or even `6x30s` - output will be less cluttered.

But it depends on what / how you want to wait, just an idea...